blocks/dfio.py

```diff
@@ -33,34 +29,23 @@ def read_df(datafile, **read_args):
     -----
     The read functions are taken from pandas, e.g. pd.read_csv
     Check the pandas doc for more information on the supported arguments
-
     """
-    filetype, compression = _get_extension(datafile.path)
+    filetype = _get_extension(path)
     reader = _readers[filetype]
-    if (
-        reader in (pd.read_csv, pd.read_pickle, pd.read_json)
-        and compression is not None
-    ):
-        read_args["compression"] = compression
-    elif reader in (pd.read_csv, pd.read_pickle, pd.read_json):
-        read_args[
-            "compression"
-        ] = None  # default "infer" incompatible with handles as of 0.24
     if reader == pd.read_json:
         # Default json file is newline delimited json records, but can be overwritten
         defaults = {"lines": True, "orient": "records"}
         defaults.update(read_args)
         read_args = defaults
 
-    with datafile.handle("rb") as f:
-        return reader(f, **read_args)
+    return reader(path, **read_args)
 
 
-def write_df(df, datafile, **write_args):
+def write_df(df, path, **write_args):
     """Write a dataframe to file based on the file extension
 
     The following formats are supported:
-    parquet, avro, csv, pickle
+    parquet, avro, csv, pickle, json
```

Comment:
would be nice to have a note on which types are supported. reading the code, should be any writer supported by pandas (i.e. `pandas.DataFrame.to_xyz`), with some possibly additional glue code.